scripts/setup_automation.py

```
@@ -0,0 +1,597 @@
+            for hook_type in ['pre-commit', 'post-commit', 'pre-push']:
+                hook_file = self.git_manager.hooks_dir / hook_type
+                if hook_file.exists():
+                    self.logger.info(f"✅ {hook_type} hook installed")
+                    checks.append(True)
+                else:
+                    self.logger.warning(f"⚠️ {hook_type} hook not found")
+                    checks.append(False)
+        
+        # Check if components can be imported
+        try:
+            from src.automation import AutoDocsWatcher, DocumentationGenerator, GitHooksManager
```

Ruff (0.11.9)

`src.automation.AutoDocsWatcher` imported but unused; consider using `importlib.util.find_spec` to test for availability

(F401)